pkg/logging/logging.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 The Kubernetes Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package logging contains helper code for initializing logging the the
+// commands.
+package logging
+
+import (
+	"strings"
+
+	"github.com/spf13/pflag"
+	"github.com/urfave/cli/v2"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/component-base/featuregate"
+	logsapi "k8s.io/component-base/logs/api/v1"
+
+	_ "k8s.io/component-base/logs/json/register" // for JSON log output support
+)
+
+// Flags creates a new logging configuration. It returns the function which
+// applies that configuration and the corresponding flags.
+func Flags() (before func(*cli.Context) error, flags []cli.Flag) {
+	featureGate := featuregate.NewFeatureGate()
+	utilruntime.Must(logsapi.AddFeatureGates(featureGate))
+	utilruntime.Must(featureGate.SetFromMap(map[string]bool{string(logsapi.ContextualLogging): true}))
+	logsconfig := logsapi.NewLoggingConfiguration()
+	var fs pflag.FlagSet
+	logsapi.AddFlags(logsconfig, &fs)
+
+	// Adding the feature gates flag to fs means that its going to be added
+	// with "logging" as category. In practice, the logging code is the
+	// only code which uses the flag, therefore that seems like a good
+	// place to report it.
+	fs.AddFlag(&pflag.Flag{
+		Name: "feature-gates",
+		Usage: "A set of key=value pairs that describe feature gates for alpha/experimental features. " +
+			"Options are:\n     " + strings.Join(featureGate.KnownFeatures(), "\n     "),
+		Value: featureGate,
+	})
+
+	fs.VisitAll(func(flag *pflag.Flag) {
+		flags = append(flags, pflagToCLI(flag, "Logging:"))
+	})
+
+	return func(*cli.Context) error {
+		// Activate logging as soon as possible.
+		return logsapi.ValidateAndApply(logsconfig, featureGate)
+	}, flags
+}
+
+func pflagToCLI(flag *pflag.Flag, category string) cli.Flag {
```

Comment:
I found `App.AllowExtFlags`:
> Allows global flags set by libraries which use flag.XXXVar(...) directly
> to be parsed through this library

However, that implies adding logging flags to `flag.Cmdline`, which doesn't feel right. I prefer this solution.